src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,224 @@
+            String currLine = reader.readLine();
+
+            // Read through all saved tasks
+            while (currLine != null) {
+                String[] taskString = currLine.split(" \\| ", 3);
+                Task newTask;
+
+                // Current task is a Deadline
+                if (taskString[0].equals("D")) {
+                    String[] deadlineDetails = taskString[2].split(" ");
+                    if (deadlineDetails.length == 2) {
+                        newTask = new Deadline(deadlineDetails[0], LocalDate.parse(deadlineDetails[1]));
+                    } else if (deadlineDetails.length == 3) {
+                        newTask = new Deadline(deadlineDetails[0],
+                                LocalDate.parse(deadlineDetails[1]), LocalTime.parse(deadlineDetails[2]));
+                    } else {
+                        throw new DukeException(DukeExceptionType.DB_READ);
+                    }
+
+                // Current task is an Event
+                } else if (taskString[0].equals("E")) {
+                    String[] periodDetails = taskString[2].split(" ");
+                    if (periodDetails.length == 3) {
+                        newTask = new Event(periodDetails[0],
+                                LocalDate.parse(periodDetails[1]), LocalDate.parse(periodDetails[2]));
+                    } else if (periodDetails.length == 4) {
+                        newTask = new Event(periodDetails[0], LocalDate.parse(periodDetails[1]),
+                                LocalTime.parse(periodDetails[2]), LocalTime.parse(periodDetails[3]));
+                    } else if (periodDetails.length == 5) {
+                        newTask = new Event(periodDetails[0],
+                                LocalDate.parse(periodDetails[1]), LocalTime.parse(periodDetails[2]),
+                                LocalDate.parse(periodDetails[3]), LocalTime.parse(periodDetails[4]));
+                    } else {
+                        throw new DukeException(DukeExceptionType.DB_READ);
+                    }
+
+                // Current task is a Todo
+                } else if (taskString[0].equals("T")) {
+                    newTask = new Todo(taskString[2]);
+
+                // Current task is in invalid format
+                } else {
+                    throw new DukeException(DukeExceptionType.DB_READ);
+                }
+
+                // Handles whether or not Task is done
+                if (Integer.parseInt(taskString[1]) == 1) {
+                    newTask.setDone();
+                }
+
+                // Add Task to task list, continue with next Task
+                savedTasks.add(newTask);
+                currLine = reader.readLine();
+            }
+
+        } catch (DukeException e) {
+            ui.showException(e);
+
+        } catch (IOException e) {
+            ui.showException(new DukeException(DukeExceptionType.DB_LAUNCH));
+        }
+
+        return savedTasks;
+    }
```

Comment:
Maybe can consider adding helper methods to make this method shorter instead of having too many nested if-else blocks?